### ISSUE TITLE
How can anchore match kubernetes pullables for repositories with arch manifests?

### ISSUE THREAD

**Issue by bs-matil** [anchore/anchore-engine#91]
**Is this a request for help?**:
yes
---

**FEATURE REQUEST** :
anchore-cli, version 0.2.4
anchore-core, version 0.2.4

**What happened**:
scanning image docker.io/library/ubunut:latest
result -> 
`docker.io/library/ubuntu:latest        cd6d8154f1e16e38493c3c2798977c5e142be5e5d41403ca89883840c6d51762        analyzed
`

kubernetes status imageid shows
`sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378` 

Both ids differ and can't be matched
`sha256:cd6d8154f1e16e38493c3c2798977c5e142be5e5d41403ca89883840c6d51762`
vs
`sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378`     

**What did you expect to happen**:

You can find all corresponding sha's in the following file https://github.com/docker-library/repo-info/blob/47cef7a0293d3165dc3cd6abe85d4c5a41b84f30/repos/ubuntu/remote/18.04.md 

If you scan an official repository which uses manifest files on the root level like ubuntu you get an id for the manifest. Then you fetch the image for your architecture e.g. amd64

On kubernetes only the manifest file is stored as final status e.g. `sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378`        . 
so if you look on your image status you will find this. 

if you look in anchore for the scanned image you will only find the arch amd64 sha `sha256:cd6d8154f1e16e38493c3c2798977c5e142be5e5d41403ca89883840c6d51762`.

**Suggestion:**
anchore should also accept the arch as argument e.g. amd64 and/or should allow to use manifest ids for the image api
examples:
```
# use architecture
anchore-cli image --arch=amd64 vuln sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378 os 
[print results for amd64]

# use manifest id
anchore-cli image  vuln sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378 os 
[print results for amd64,arm,etc or an error with available arch]
```

**Comment by zhill** [comment]
Hi @bsukramlitt , thanks for using anchore! Yes, unfortunately this is a known issue due to how Docker handles and presents image digests for multi-arch images. We'll be adding better multi-arch support soon since it is becoming much more common to see now that many official images have added it. I'll look into a workaround, but right now I don't see a simple one. Given that this makes k8s integration harder, we'll prioritize this issue higher since you can't get the arch-specific digest from k8s. I'll keep this issue updated as we go since this will probably require some db updates on the anchore side.

Your suggestion is inline with what we have been thinking (add explicit arch tracking and options in cli), but we don't have a firm timeline for that change just yet. I think it would make sense to have the CLI and API default to using amd64 unless explicitly set.

**Comment by zhill** [comment]
Hi @bmatil , this issue was addressed in 0d434cba7fd283209b37239a3104b316ff1c59c9, there is now a way to resolve the digest you get from docker by matching against the digest or "parent_digest" for an image in anchore. We'll have more info in docs and release notes for the upcoming release that will contain this change, but it has been addressed and is in master branch now.

**Comment by zhill** [comment]
Closing this due to inactivity and age. If there are more questions, please re-open or file a new issue. Thanks!
